src/main/java/org/springframework/session/data/mongo/SessionConverterProvider.java

```diff
@@ -30,13 +28,7 @@ private SessionConverterProvider() {
 	}
 
 	static AbstractMongoSessionConverter getDefaultMongoConverter() {
-		
-		if (ClassUtils.isPresent(JACKSON_CLASS_NAME, null)) {
```

Comment:
@gregturn I know this has been removed, but for future reference I would not update the converter based on the classpath. There are places in Spring where it makes sense to adjust defaults based upon the classpath, but because Spring Session is persistent it is a bit different. For example, consider an application that starts off without jackson on the classpath and uses Spring Session MongoDB. Those sessions would be persisted with `JdkMonogoSessionConverter`. In the next release, they add Jackson to their classpath for a new feature. Now `JacksonMongoSessionConverter` is used which will be unable to read sessions from the previous state of the application. This might be tricky to find before moving into production since it will only impact sessions that still are not expired, so you need active sessions before this becomes a problem.

Reply:
For starters, I think this feature was already coded. I agree that autoselecting based on classpath seems kind of weird. Smells like something to only consider in a Boot autoconfiguration scenario, not here.

We definitely need suitable integration testing to support either migrating existing sessions should the developer decided to change converters (or give them the option to delete instead). But this should be a strategy chosen by the developer, not autoselected by Spring Session MongoDB.

And this only applies after ensuring JacksonMongoSessionConverter works properly.